clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractMembershipManager.java

```diff
@@ -851,6 +851,11 @@ void maybeReconcile() {
                 revokedPartitions
         );
 
+        // Mark partitions as pending revocation to stop fetching from the partitions (no new
+        // fetches sent out, and no in-flight fetches responses processed).
+        System.out.println("revokedPartitions = " + revokedPartitions);
+        markPendingRevocationToPauseFetching(revokedPartitions);
```

Comment:
I think it's better to keep `markPendingRevocationToPauseFetching` in `AbstractMembershipMember#revokePartitions`, because `ConsumerMembershipManager#invokeOnPartitionsRevokedOrLostToReleaseAssignment` also uses the function. 

Although it's redundant for consumer heartbeat response path, it still required for `ConsumerMembershipManager#signalMemberLeavingGroup` path.

Reply:
makes sense.